pkg/chaosdaemon/jvm_server_backup.go

```diff
@@ -0,0 +1,209 @@
+// Copyright 2021 Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package chaosdaemon
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/chaos-mesh/chaos-mesh/pkg/chaosdaemon/util"
+
+	"github.com/chaos-mesh/chaos-mesh/pkg/bpm"
+	pb "github.com/chaos-mesh/chaos-mesh/pkg/chaosdaemon/pb"
+	"github.com/golang/protobuf/ptypes/empty"
+)
+
+const (
+	bytemanHome            = "/usr/local/byteman/"
+	bmInstallCommandBackUp = bytemanHome + "bin/bminstall.sh -b -Dorg.jboss.byteman.transform.all -Dorg.jboss.byteman.verbose -Dorg.jboss.byteman.compile.to.bytecode -p %d %d"
+	bmSubmitCommandBackUp  = bytemanHome + "bin/bmsubmit.sh -p %d -%s %s"
+)
+
+func (s *DaemonServer) InstallJVMRulesBackUp(ctx context.Context,
+	req *pb.InstallJVMRulesRequest) (*empty.Empty, error) {
+	pid, err := s.crClient.GetPidFromContainerID(ctx, req.ContainerId)
+	if err != nil {
+		log.Error(err, "GetPidFromContainerID")
+		return nil, err
+	}
+
+	// todo: Need to write the BYTEMAN_HOME environment variable in bminstall.sh and bmsubmit.sh
+	// or do this in code ?
+	agentFile, err := os.Open("/usr/local/bin/byteman.tar.gz")
+	if err != nil {
+		return nil, err
+	}
+	//processBuilder = bpm.DefaultProcessBuilder("sh", "-c", "cat > /usr/local/byteman/lib/byteman.jar").SetContext(ctx)
+	processBuilder := bpm.DefaultProcessBuilder("sh", "-c", "cat > /usr/local/byteman.tar.gz").SetContext(ctx)
+	processBuilder = processBuilder.SetNS(pid, bpm.MountNS).SetStdin(agentFile)
+	output, err := processBuilder.Build().CombinedOutput()
+	if err != nil {
+		return nil, err
+	}
+	if len(output) > 0 {
+		log.Info("copy byteman.tar.gz", "output", string(output))
+	}
+	// uncompress byteman.tar.gz
+	processBuilder = bpm.DefaultProcessBuilder("sh", "-c", "tar -zxf /usr/local/byteman.tar.gz -C /usr/local").SetContext(ctx).SetNS(pid, bpm.MountNS)
+	output, err = processBuilder.Build().CombinedOutput()
+	if err != nil {
+		return nil, err
+	}
+	if len(output) > 0 {
+		log.Info("tar byteman.tar.gz", "output", string(output))
+	}
+
+	// Assuming the first process is the java process
+	bmInstallCmd := fmt.Sprintf(bmInstallCommandBackUp, req.Port, 1)
```

Comment:
could find the first java process, refer to https://github.com/chaos-mesh/chaos-mesh/pull/2904

Reply:
I have studied this code, and all he looks for is the pid of the host where the chaos-daemon is located, but here byteman needs to attach the pid inside the container, and needs to read the /proc directory inside the container